sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/BigQueryTableInserter.java

```diff
@@ -273,6 +296,137 @@ public void insertAll(TableReference ref, List<TableRow> rowList,
   }
 
   /**
+   * Import files into BigQuery with load jobs.
+   */
+  public void load(
+      String jobId,
+      TableReference ref,
+      List<String> gcsUris,
+      TableSchema schema,
+      WriteDisposition writeDisposition,
+      CreateDisposition createDisposition) throws InterruptedException, IOException {
+    JobConfigurationLoad loadConfig = new JobConfigurationLoad();
+    loadConfig.setSourceUris(gcsUris);
+    loadConfig.setDestinationTable(ref);
+    loadConfig.setSchema(schema);
+    loadConfig.setWriteDisposition(writeDisposition.name());
+    loadConfig.setCreateDisposition(createDisposition.name());
+    loadConfig.setSourceFormat("NEWLINE_DELIMITED_JSON");
+
+    String projectId = ref.getProjectId();
+    for (int i = 0; i < MAX_RETRY_LOAD_JOBS; ++i) {
+      String retryingJobId = jobId + "-" + i;
+      insertLoadJob(retryingJobId, loadConfig);
+      Status jobStatus = pollJobStatus(projectId, retryingJobId);
+      if (jobStatus == Status.SUCCEEDED) {
+        return;
+      } else if (jobStatus == Status.UNKNOWN) {
+        throw new RuntimeException("Failed to poll the load job status.");
+      }
+      // continue for Status.FAILED
+    }
+    throw new RuntimeException(
+        "Failed to create the load job, reached max retries: " + MAX_RETRY_LOAD_JOBS);
+  }
+
+  @VisibleForTesting
+  void insertLoadJob(String jobId, JobConfigurationLoad loadConfig)
```

Comment:
The failure handling here is very complex. For everyone one of these functions, I'd like a clear Javadoc saying what it does, what can go wrong, how this is handled, and what the caller needs to do.